### ISSUE TITLE
Add ability to use first read separately in NIRCam imaging

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#8836]
_Issue [JP-3766](https://jira.stsci.edu/browse/JP-3766) was created on JIRA by [Timothy Brandt](https://jira.stsci.edu/secure/ViewProfile.jspa?name=tbrandt):_

NIRCam is capable of downlinking the first read separately even when the first group includes the first read in an average.  For example, in DEEP8 mode, the first group consists of the first 8 reads, but the first read itself may be individually available in the uncalibrated data file.  This ticket is to make use of that first read in ramp fitting and jump detection.

There are a few reasons why this is valuable:
 # For a source that saturates after the first group.  With the first read separated out, there are now two groups that can be fit rather than just one.  
 # If there are only four groups, then jump detection becomes very difficult.  For example, if a cosmic ray hits in the middle of group 2 of a four group ramp, then the differences between group 2 and group 1 and between group 3 and group 2 are both bad; only the difference between group 4 and group 3 is valid.  This is very difficult to handle in jump detection.  If the first read can be separated out, we then have five groups and can almost fully recover the data.

The approach to separating out the first read is described in

<https://ui.adsabs.harvard.edu/abs/2024PASP..136d5005B/abstract>

In short, if we have a list of lists for the read times that comprise each group, e.g.

readtimes = [[1, 2, 3, 4], [9, 10, 11, 12], [17, 18, 19, 20], [25, 26, 27, 28]]

then we would like to modify this with something like

readtimes = [readtimes[0][0]] + [readtimes[0][1:]] + readtimes[1:]

to make

readtimes = [[1], [2, 3, 4], [9, 10, 11, 12], [17, 18, 19, 20], [25, 26, 27, 28]]

We would also need to remove the first read from the first group.  In this case, that means the first group becomes

group1_new = 1/3*(group1_old*4 - read1)

and the first read would get prepended to the groups.

**Comment by stscijgbot-jp** [comment]
_Comment by [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) on [JIRA](https://jira.stsci.edu/browse/JP-3766?focusedCommentId=731408#comment-731408):_

Clarifying that this is for use in the likelihood-based ramp fitting algorithm.  Ticket is ON HOLD until successful resolution of <https://jira.stsci.edu/browse/JP-3765>

**Comment by stscijgbot-jp** [comment]
_Comment by [Tyler Pauly](https://jira.stsci.edu/secure/ViewProfile.jspa?name=tpauly) on [JIRA](https://jira.stsci.edu/browse/JP-3766?focusedCommentId=953326#comment-953326):_

Ticket has been duplicated by JP-3891 - because it has more recent activity, perhaps we can close this one.

**Comment by stscijgbot-jp** [comment]
_Comment by [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) on [JIRA](https://jira.stsci.edu/browse/JP-3766?focusedCommentId=953334#comment-953334):_

Hm, agreed.
